rasa/nlu/selectors/response_selector.py

```diff
@@ -904,6 +954,15 @@ def batch_predict(
             batch_in, self.predict_data_signature
         )
 
+        # Edge Case: In some cases, the model_data preparation will have
+        # skipped sequence-level features because they are not needed for
+        # the architecture.
+        tf_batch_data[TEXT].setdefault(SEQUENCE, [])
```

Comment:
Now that I see this being done in 3 different functions, it makes me wonder can't we set a default value for `SEQUENCE` inside the batch_data creation itself?

Reply:
Otherwise, we have duplicated code and if someone tries to inherit from these classes for a custom component, they are bound to not notice this 😅

Reply:
It was because of the following lines (see comment below). Even though we set this to [] / could catch a None for this later, we still need the SEQUENCE_LENGTH == 0 to make this work. ... Well, we could also catch that later. Hmm. Will try to simplify this.

Reply:
Ok, I won't because: The root cause is how add_features for RasaModelData works. If it wouldn't silently remove stuff, then things would be easier (and those empty lists / 0's I set there would end up in the final tf_batch_data, for example). But I won't touch that function now.

I think to resolve having to think of the edge case it would be great to simply give the special case with 0 transformer layers a different name. It seems to make little sense to use that for entity recognition (?) and it is collapsing so much it seems that baseline deserves its own name. What do you think?

Apart from that, we'll look into add_features / RasaModelData anyway with the refactorings coming up, right?

Reply:
Okay, all of that makes sense 💯 . Can you please create an issue with all this info so that we are able to keep a track of it? I would be supppper eager to remove these one liners that are duplicated in all functions just to support an edge case.

Reply:
Btw, just for completeness: in some cases it is - strictly speaking not necessary to add that setdefault - because `batch_to_model_data_format` returns a `defaultdict`. But it is very confusing just to leave it out there and rely on the side effect that I actually would not have expected from `batch_to_model_data_format` :/

Reply:
resolving this one - replaced the setdefault approach by removing check in get_sequence_length, renaming those, and adding comments on defaultdicts to avoid "surprising" us again